Remove shop links from plugin details modal

mneudert opened the pull request:
### Description:

Removes the CTA buttons leading to the webshop from the plugin detail modals.

The case fallback for `{% if plugin.shop.variations|length %}` was also removed for consistency with the CTA display behaviour. To my knowledge it should not happen that a premium plugin has no variations, and the links are still available in the description part of the modal.

Whitespace above the description has been adjusted to be a general "4 br height if not cta visible", though yet to be checked if it will be adjusted to a slightly lower amount.

Refs DEV-18004

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

mneudert committed:
Remove shop links from plugin details modal

michalkleiner commented:
The space above the metadata looks weird without the CTA, I've tried to mock a quick tweak that would make it look slightly better — add the below CSS to `.metadata` and remove the artificial line breaks creating the space.
```
div.metadata {
    border-left: 1px solid silver;
    padding-left: 10px;
    margin-left: 10px;
}
```
With the above CSS and with line breaks on the left, with a CTA that stays on the right — what do you think?
![Screenshot 2024-04-24 at 6 26 39 PM](https://github.com/matomo-org/matomo/assets/233342/85f1ad86-976c-480a-8af9-e24c769c3991)

Without the CSS the alignment of the metadata column is just weird (even though it's what it was before horizontally).
![Screenshot 2024-04-24 at 6 33 12 PM](https://github.com/matomo-org/matomo/assets/233342/9e97e6a6-f977-4866-9624-9b6999a6dc34)

mneudert commented:
The margin/padding is more or less already taken care of in #22055 with the removal of `.pluginDetails > .row > .col` there. If we change that here it should be the same change to have it change only once.

Displaying a silver border looks somewhat odd to me if it does not match the whole height, so i am inclined to skip that part :thinking: 

Pushed a matching change so we can see what it looks like after CI has generated the screenshot diffs :eyes:

michalkleiner commented:
We could put the border (silver or light gray) on the content, so it would be the whole height.
The generated UI screenshots seem to make the gap quite big, perhaps make it a bit smaller? Also we may need to add right padding as now the content is touching the border of the modal.

I'd say whichever way is the easiest as long as we keep the metadata at the top and it's somewhat visually separated from the content.

mneudert commented:
I will add one or two changes so we (or probably you) can check with Javi on Monday which screenshot set we should keep.

To me that sounds like a reasonable way to invest the time, instead of doing everything without prior UI approval :see_no_evil:

michalkleiner commented:
Good call 👍

mneudert committed:
Tweak whitespace usage

mneudert committed:
Add silver lining to metadata

mneudert committed:
Extend silver lining to content

mneudert commented:
@michalkleiner created 4 sets of screenshots we could get checked:

- [first commit](https://builds-artifacts.matomo.org/matomo-org/matomo/dev-18004/8837473996/): `<br>` tags
- [second commit](https://builds-artifacts.matomo.org/matomo-org/matomo/dev-18004/8846431682/): removed `<br>` tags and added margin/padding
- [third commit](https://builds-artifacts.matomo.org/matomo-org/matomo/dev-18004/8846900626/): added the silver border to the metadata block
- [fourth commit](https://builds-artifacts.matomo.org/matomo-org/matomo/dev-18004/8847327331/): expanded silver border to full content height

The full height border will only be visible for actual modals, the UI tests don't have enough content for it to show. And it should only be checked with a plugin that displays images in the description, as those images may touch or even overlap the border.

Metadata content can currently overflow outside of the modal. If a link is displayed that contains no characters where a browser would break the line it will flow out of the metadata block. `.Piwik_Popover` already provides padding so the metadata content could be set to `overflow: hidden` to avoid that part.

michalkleiner commented:
Spoke about this with the design team today and the preference is full height silver line, no space above the metadata (there's still an empty paragraph from somewhere) and if possible, a bit of padding so that the images don't touch the vertical line (which can be achieved by removing flex 1 and making them 48% wide).

michalkleiner committed:
Prevent metadata overflow when words are long and don't have natural break points

michalkleiner committed:
Display vertical break only when there's a CTA button or a special offer available

michalkleiner committed:
Show right-aligned images in plugin content readme at 48% width to avoid colision with the newly added border

michalkleiner commented:
@mneudert pushed two commits based on the discussion with design, could you check it looks ok from your perspective and update the test screenshots, please?

mneudert committed:
Completely remove whitespace above old plugin details metadata

mneudert committed:
Override inconsistent marketplace "*-div-readme" styling

mneudert commented on plugins/Marketplace/stylesheets/plugin-details.less:
```diff
@@ -43,9 +43,25 @@
     }
 
+    // some plugins receive inline style from the marketplace
+    // override them for consistency
```
The adjustment originally done in 6e02b6f9c1d52ee539e16e449bf4d28bde31ee27 targeted the new modal design.

Copied the full definition here with the flex/width adjustment. Most plugins receive the styling as inline in the marketplace "documentation" part (e.g. Funnels), though some don't (e.g. Cohorts). Now we should only check if that is a complete adjustment or if it breaks something unintended :frowning_face:

michalkleiner replied:
@mneudert spoke with design on this yesterday and the longterm plan is to adjust the content to remove the left/right mechanism and side by side content with images as much as possible to make it better legible, so I think we don't need to spend too much time on that. I haven't found any other use of the styles within the codebase, so chances are high this is ok for now.

mneudert committed:
Update expected screenshots

michalkleiner commented:
When I woke up today I had the same idea to remove the empty paragraph and add the margin-bottom to the action div after I saw the UI test screenshots, but I ran out of time on that, so well done you've done the same! :)
